File: java/learning-spring-01/src/main/java/com/example/learningspring/beans/helloworld/App02HelloWorldSpring.java
package com.example.learningspring.beans.helloworld;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;

import java.util.Arrays;

public class App02HelloWorldSpring {
    public static void main(String[] args) {
        // 1. Spring Context 가져오기 -
        try (var context = new AnnotationConfigApplicationContext(HelloWorldConfiguration.class)) {
            // 2. Spring이 관리하길 원하는 걸 정하기 - @Configuration
            System.out.println(context.getBean("name"));
            System.out.println(context.getBean("age"));
            System.out.println(context.getBean("person"));
            System.out.println(context.getBean("person2MethodCall"));
            System.out.println(context.getBean("person3Parameters"));
            System.out.println(context.getBean("person5Qualifier"));
            System.out.println(context.getBean("address2"));

            // error
            System.out.println(context.getBean(Person.class));
            // address2와 address3 중 어떤 걸 선택해야하는 지를 몰라 에러가 발생한다.
            // System.out.println(context.getBean(Address.class));

            // System.out.println은 안되는 이유?
            Arrays.stream(context.getBeanDefinitionNames()).forEach(System.out::println);
        }
    }
}

File: kotlin/learning-spring/src/main/kotlin/com/examplekotlin/learningspring/beans/helloworld/App02HelloWorldSpring.kt
package com.examplekotlin.learningspring.beans.helloworld

import org.springframework.context.annotation.AnnotationConfigApplicationContext


fun main(args: Array<String>) {

    // 왜 java class를 넣어줘야 하지?
    // Kotlin에서 class는 KClass를 Return
    // class.java는 Class를 Return.
    // Type이 다르기 때문에 class만 넣으면 type error
    AnnotationConfigApplicationContext(HelloWorldConfiguration::class.java).use { context ->
        println(context.getBean("name"));
        println(context.getBean("age"))
        println(context.getBean("person"))
        println(context.getBean("person2MethodCall"))
        println(context.getBean("person3Parameters"))
        println(context.getBean("address2"))
    }


}